Incorporate GAT feedback (#1922)
galaxyproject/training-material: topics/admin/tutorials/ansible-galaxy/tutorial.md

>    5. `check_migrate_tools` must be set to `false` due to a new installation of Galaxy.
>    6. `tool_data_path` to `{{ galaxy_mutable_data_dir }}/tool-data`, so that when tools are installed, due to privilege separation, this will happen in a directory Galaxy can actually write into.
>
>    > ### {% icon tip %} Data storage
>    > Galaxy datasets cannot be separated by user or other attribute currently. Currently you can only spread data unintelligently across 1 or more storage pools.
>    {: .tip}
>

> >    > ### {% icon tip %} PostgreSQL Connection String

Reviewer: ```suggestion
>    > ### {% icon tip %} PostgreSQL connection string
```

>    ```
>    {% endraw %}
>
>    > ### {% icon tip %} How many mules?
>    > Start with 1 and add more as needed. If you notice that your jobs seem to inexplicably sit for a long time before being dispatched to the cluster, or after they have finished on the cluster, you may need additional handlers.
>    {: .tip}
>

> >    > ### {% icon tip %} uWSGI Threads, Offload Threads, Mules, Etc.

Reviewer: Maybe this tip should be moved before the "How many mules?" one?

```suggestion
>    > ### {% icon tip %} uWSGI threads, offload threads, mules, etc.
```

>
{: .comment}

# Production & Maintenance

> This depends on the number of users and their specific needs, but a smallish server (<= 25 users) will typically require a day or two per month of maintenance. Large public servers like usegalaxy.org and usegalaxy.eu are largely full time jobs (although even their admins do find time to do other things).

Reviewer: ```suggestion
The time required to maintain a production Galaxy instance depends on the number of users and their specific needs, but a smallish server (<= 25 users) will typically require a day or two per month of maintenance. Large public servers like usegalaxy.org and usegalaxy.eu are largely full time jobs (although even their admins do find time to do other things).
```

## User Support

There are many user support resources available to you and your users online. [help.galaxyproject.org](https://help.galaxyproject.org) is the primary landing point for Galaxy users. For helping individual users, you might find it useful to impersonate users if they fail to send adequate bug reports.

### Impersonating

You can activate user impersonation with

```
allow_user_impersonation: true
```

We (admins) generally ask permission or consent from the user "Hey, mind if we look at your history", and then avoid running things in their history. It can confuse users when datasets show up unexpectedly, since Galaxy is not normally a real-time collaborative activity. Additionally you can automatically send failing job error reports, even if users do not click "subimt", and maybe proactively address those issues. EU doesn't do this due to the volume of issues.

## Running on a cluster

If you need to run on a cluster with a shared file system, you will need to expose several directories to your cluster:

- `galaxy_shed_tools_dir`
- `galaxy_tool_dependency_dir`
- `galaxy_file_path`
- `galaxy_job_working_directory`
- `galaxy_server_dir`
- `galaxy_venv_dir`

> Some of these can be worked around, by running the portions of the roles that deploy these directories, on the shared filesystem. Then Galaxy and the shared filesystem can run off of two difference copies of them, if that is better for performance:

Reviewer: ```suggestion
Some of these can be worked around, by running the portions of the roles that deploy these directories on the shared filesystem. Then Galaxy and the shared filesystem can run off of two difference copies of them, if that is better for performance:
```
